File: CSharp/Ds_and_algo/BubbleSort.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Ds_and_algo
{
    class BubbleSort
    {
        public int[] arr;
        public BubbleSort(int[] _arr)
        {
            arr = _arr;
        }

        public void Sort()
        {
            int arrLength = arr.Length - 1;
            int temp;
            for (int i = 0; i < arr.Length; i++)
            {
                for (int j = 0; j < arrLength; j++)
                {
                    if (arr[j] > arr[j + 1])
                    {
                        temp = arr[j];
                        arr[j] = arr[j + 1];
                        arr[j + 1] = temp;
                    }
                }

                arrLength--;
            }
            Console.WriteLine($"Sorted array:- {string.Join(", ", arr)}");

        }
    }
}

File: javascript/BubbleSort.js
// Continuous comparison with neighbours will bubble sorted elements
function BubbleSort(arr) {
  let arrLength = arr.length - 1;
  let temp;
  for (let i = 0; i < arr.length; i++) {
    for (let j = 0; j < arrLength; j++) {
      if (arr[j] > arr[j + 1]) {
        temp = arr[j];
        arr[j] = arr[j + 1];
        arr[j + 1] = temp;
      }
    }
    arrLength--;
  }

  console.log(`Sorted array:- ${arr}`);
}

BubbleSort([345, 51, 1, 99, 7, 3, 5, 7, 46, 34, 76, 22, 78, 34, 78, 98]);
